index.bs

```diff
@@ -8626,28 +8613,39 @@ This is useful in cases where a context identifier can stand in for the realm
 associated with the context's active document.
 
 <div algorithm>
-To <dfn>get a realm from a target</dfn> given |target|:
+To <dfn>get a realm from a browsing context</dfn> given |context id| and |sandbox|:
 
-1. If |target| matches the <code>script.ContextTarget</code> production:
+1. Let |context| be the result of [=trying=] to [=get a browsing context=]
+   with |context id|.
 
-  1. Let |context| be the result of [=trying=] to [=get a browsing context=]
-     with |target|["<code>context</code>"].
+1. If |sandbox| is null or is an empty string:
 
-  1. If |target| does not contain a field named "<code>sandbox</code>", or
-     |target|["<code>sandbox</code>"] is an empty string:
+  1. Let |document| be |context|'s [=active document=].
 
-    1. Let |document| be |context|'s [=active document=].
+  1. Let |environment settings| be the [=environment settings object=] whose
+     [=relevant global object=]'s <a>associated <code>Document</code></a> is
+     |document|.
 
-    1. Let |environment settings| be the [=environment settings object=] whose
-       [=relevant global object=]'s <a>associated <code>Document</code></a> is
-       |document|.
+  1. Let |realm| be |environment settings|' [=realm execution context=]'s
+     Realm component.
+
+1. Otherwise: let |realm| be result of [=trying=] to
+   [=get or create a sandbox realm=] given |sandbox| and
+   |context|.
+
+1. Return [=success=] with data |realm|
+
+Issue: This has the wrong error code
+</div>
+
+<div algorithm>
+To <dfn>get a realm from a target</dfn> given |target|:
 
-    1. Let |realm| be |environment settings|' [=realm execution context=]'s
-       Realm component.
+1. If |target| matches the <code>script.ContextTarget</code> production:
 
-  1. Otherwise: let |realm| be result of [=trying=] to
-     [=get or create a sandbox realm=] given |target|["<code>sandbox</code>"] and
-     |context|.
+  1. Let |realm| be [=get a realm from a browsing context=] with
```

Comment:
I'd prefer to split this into two steps, first getting the sandbox and then calling `get a realm from a browsing context`, just to avoid the awkward parsing of the "if present, or null otherwise" part.

Reply:
Agree, done